compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -1386,8 +1386,9 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
         };
 
         self.note_obligation_cause(&mut err, &obligation);
-        self.point_at_returns_when_relevant(&mut err, &obligation);
-
+        if self.point_at_returns_when_relevant(&mut err, &obligation) {
+            err.delay_as_bug();
+        }
```

Comment:
To mirror what compiler-errors said, here you could instead check if `obligation.references_error()`, instead of changing `point_at_returns_when_relevant`.

Reply:
I tried, but it didn't work, thus `obligation.predicate.references_error()` behaves differently from` ty_references_error()`.
I've changed `point_at_returns_when_relevant()` as Michael suggested, which works well.